multiselect/add-new-item/MultiSelectAndAddButton/MultiSelectAndAddButton/Shared/MultiSelectWithAdd.razor

```diff
@@ -50,17 +61,31 @@
     async Task OnCreateRecord()
     {
         OptionsModel insertedOption = await _dataService.AddOption(_currentFilterValue);
-        Options.Add(insertedOption); // update the current view-model. Add() works because it is an observable collection
 
-        // update the values, needs a new reference, see https://docs.telerik.com/blazor-ui/common-features/observable-data#refresh-data
-        // continues with a hack in OnRead
+        // continues with a hack in OnRead and OnAfterRenderAsync
         Tags.Add(insertedOption.MyValueField);
-        shouldUpdateValueParam = true;
+        shouldClearInput = true;
+
+        MultiSelectWithButton.Rebind();
 
         // raises event for parent componets to use
         await TagsChanged.InvokeAsync(Tags);
     }
 
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (shouldUpdateTags)
+        {
+            shouldUpdateTags = false;
+
+            // hack to refresh the selected values AFTER the MultiSelect is rebound
+            Tags = new List<int>(Tags);
+            StateHasChanged();
```

Comment:
Do we need the `tateHasChanged`. I believe after the Tags is set, the Multiselect will rerender since it has updated param. Do we need to update the whole page?

Reply:
Yes, with the current implementation `StateHasChanged` is required even if I reset the Tags instance. I wondered about this too.